lib/core/src/Cardano/Wallet/Util.hs

```diff
@@ -167,3 +173,47 @@ parseURI (T.unpack -> uri) = runIdentity $ runExceptT $ do
                     Left "URI must not contain a path/query/fragment."
             _ -> Right uri'
     either (throwE . TextDecodingError) pure res
+
+--------------------------------------------------------------------------------
+-- Containers
+--------------------------------------------------------------------------------
+
+-- | A container that always has no elements.
+--
+-- This type is the conceptual opposite of 'NonEmpty'.
+--
+data Empty a = Empty
+    deriving (Eq, Foldable, Functor, Ord, Show, Traversable)
+
+instance Semigroup (Empty a) where
+    Empty <> Empty = Empty
+
+instance Monoid (Empty a) where
+    mempty = Empty
+
+instance Applicative Empty where
+    pure = const Empty
+    Empty <*> Empty = Empty
+
+instance Monad Empty where
+    (>>=) = (const . const) Empty
```

Comment:
@Anviking might have a way to simplify these definitions.

In any case, we can easily test them with `quickcheck-classes`, as we do for other instances of standard classes.

Reply:
Alternatively, we can make these definitions go away if we make the following transformation:
- `Solo Coin` -> `Coin`
- `Empty Void` -> `()`

Reply:
No code is the best code!

Reply:
Fixed in b1fbdf17fa28c1126be7bfacf9360aab104a2e0b.